### ISSUE TITLE
Feast should raise an exception when trying to get a feature set that does not exist

### ISSUE THREAD

**Issue by thisisandreeeee** [gojek/feast#346]
## Expected Behavior 
Feast should raise an exception when trying to get a feature set that does not exist, even when the `fail_if_missing` parameter is specified.

## Current Behavior
Feast prints the following error message and does not raise an exception.
```
GetFeatureSet failed with code "StatusCode.UNKNOWN"
```

## Steps to reproduce
```python
from feast import Client
client = Client(core_url, serving_url)
client.get_feature_set("invalid_feature_set_name", fail_if_missing=True)
```

**Comment by woop** [comment]
@thisisandreeeee 

I am thinking of just removing this `fail_if_missing` flag and just always raising an exception here.

Would that work?

**Comment by woop** [comment]
Actually this applies to most methods. We aren't always raising a gRPC exception, but I think we should be raising that in almost every case and having you wrap it if needed.

**Comment by thisisandreeeee** [comment]
That makes sense. We can catch the exception and handle it on our end.

**Comment by woop** [comment]
The above PR fixes it #353
